libs/sdk-core/src/persist/transactions.rs

```diff
@@ -394,17 +404,9 @@ fn filter_to_where_clause(
     }
 
     if let Some(filters) = metadata_filters {
-        filters.iter().for_each(
-            |PaymentMetadataFilter {
-                 search_path,
-                 search_value,
-             }| {
-                where_clause.push(format!(
-                    "metadata->'$.{}' = '{}'",
-                    search_path, search_value
-                ));
-            },
-        );
+        filters.iter().for_each(|_| {
+            where_clause.push("metadata->'$.?'='?'".to_string());
```

Comment:
I wonder if these single quotes are the issue:
```suggestion
            where_clause.push("metadata->?=?".to_string());
```
then when building the params:
```
metadata_filters.iter().for_each(|PaymentMetadataFilter { search_path, search_value }| {
    params.push(format!("$.{}", search_path));
    params.push(search_value.clone());
})
```